test/completion.cjs

```diff
@@ -608,6 +624,37 @@ describe('Completion', () => {
       r.logs[0].should.match(/ndm flintlock >>/);
     });
 
+    describe('bash shell', () => {
+      it('generates multi-word completion handler', () => {
+        process.env.SHELL = '/bin/bash';
+        const r = checkUsage(() =>
+          yargs([]).scriptName('1 test ing').showCompletionScript()
+        );
+
+        r.logs[0].should.match(
+          /"\${COMP_WORDS\[0]}" == "1" && "\${COMP_WORDS\[1]}" == "test" && "\${COMP_WORDS\[2]}" == "ing"/
+        );
+      });
+
+      it('uses a "safe" app name for the bash completion function name', () => {
+        process.env.SHELL = '/bin/bash';
+        const r = checkUsage(() =>
+          yargs([]).scriptName('1 test ing').showCompletionScript()
+        );
+
+        r.logs[0].should.match(/_1_test_ing/);
+      });
+
+      it('replaces the bash `complete` word with only the first word of the app name', () => {
+        process.env.SHELL = '/bin/bash';
+        const r = checkUsage(() =>
+          yargs([]).scriptName('1 test ing').showCompletionScript()
+        );
+
+        r.logs[0].should.match(/_yargs_completions 1/);
+      });
+    });
+
```

Comment:
The bash shell has different replacements than the zsh version, so I test those specifically here.